### ISSUE TITLE
inconsistencies when entering a stage where individual jitsi is already on the stage

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/upstage#1312]
if you enter a stage where a player has already placed their individual jitsi webcam on the stage, you may or may not see or hear it. sometimes it works as it should - the person entering sees and hears the individual jitsi on the stage. sometimes they only hear it, and don't see anything, other times they only see it and don't hear anything.

vicki & i were testing the streams today when we observed this, & we tried re-entering the stage a number of times and in different browsers to see if we could identify any replicable behaviour, but we could not.

it is quite important that the individual jitsi is there and working properly when people enter, as there can be many situations where someone has to reload during an event, or simply arrives a little bit late. 

note - the meeting functions well, it is always there when you enter if it's already been placed on the stage. perhaps this is because it has to ask permission for cam & mic. but it's not a solution to use the meeting instead of individual jitsi as especially in performances we don't usually want to have the meeting, we want the features of individual jitsi.

**Comment by helenvarley** [comment]
testing today, two others (one logged in, one not) entering the stage where my individual stream was already on the stage, neither could see or hear it. eva, logged in, could see the frame of my stream but not the image or hear it. even when i took it off and put it on again, they could not both see and hear it.

**Comment by gloriajw** [comment]
Is this still open?

**Comment by helenvarley** [comment]
i am not sure; annie and i are planning to use the individual jitsi in our performance next week - mainly for audio stream; so i hope it is working! we tested yesterday & all good, but was only us. we're rehearsing again at 5pm CET today (in 1 hour & 45m) with 2 audience, so will note here if it works or not.

**Comment by helenvarley** [comment]
we need both, as they do different things. jitsi meet doesn't allow us to have our own streams, it always includes all of the audience as well - which is useful for discussions or if you want audience participation, but not at all useful for most performances where you just want to have your own stream.

**Comment by gloriajw** [comment]
So we haven't been successful at turning off the audience participation?

**Comment by helenvarley** [comment]
i don't know - but i don't think that's a solution; because if we're using the meeting tool without audience, it's still going to have all the jitsi tools, and perhaps also all the players, in a grid.
the beauty of the individual jitsi is that it's a clean webcam image, nothing else. just your webcam.

**Comment by gloriajw** [comment]
Right, but the moderator's window is showing on the stage, so everyone sees this view, unfortunately. 
Can the grid view be turned off??

**Comment by gloriajw** [comment]
OK, excellent. Then @helenvarley , how does jitsi meet work against you, then? Can you elaborate? Wu says the controls and grid are not visible to everyone else.

**Comment by helenvarley** [comment]
i've just had a proper look at what is on dev-app, & yes it's correct that the controls & grid aren't visible to the audience any more, however this is unfortunately not the solution that we want.
it means that we no longer have the possibility to have a discussion with the audience, & we also do not have the nice clean individual jitsi window. 
the audience view still shows the green bar at the top, my name over the image at the bottom, and if i roll over it there are tools at the top as well that show the name of the meeting, the time, and the performance settings.
as a player i can't make the image round or oval, or flip it - all things we can currently do with the individual jitsi.

**Comment by helenvarley** [comment]
one of the issues with the individual jitsi is that if someone enters the stage after a player has put their individual jitsi webcam onto the stage, the person entering doesn't see or hear it. whereas with the meeting, this is visible to people entering after it's placed on the stage.

here are two screenshots to show the difference in what the audience sees with the individual jitsi as it is now on upstage.live, and the new meeting on dev-app:
![Screenshot from 2024-09-12 18:58:14](https://github.com/user-attachments/assets/3af90260-5954-4a8d-b771-7a6ae5c504fd)
![Screenshot from 2024-09-12 18:57:14](https://github.com/user-attachments/assets/a1ba894b-3e06-4348-8722-1aae4723ae2a)

**Comment by gloriajw** [comment]
The shape and orientation of jitsi meet should be controllable. @webdev604 can you please verify this?
It sounds like there are two run modes: one where you are presenting to the audience and they do not participate in the Aactual video window, and one where they do. Is this correct?

**Comment by helenvarley** [comment]
>  It sounds like there are two run modes: one where you are presenting to the audience and they do not participate in the Actual video window, and one where they do. Is this correct?

yes - one which is an individual player's webcam and audio, and nothing else; the other where there can be audience participation.

> 
> This discussion becomes more clear if we focus on the features you need instead of what tool you want.

sorry for the confusion!

**Comment by webdev604** [comment]
ok, I will check if the shape and orientation is controllable.

**Comment by webdev604** [comment]
Now we have both Stream and Meeting and the shape and orientation should be controllable for stream.
And I checked jitsi configure but for now there is no way to hide the name of the meeting, the time, and the performance settings.
Also we need to keep the green bar at the top to be moveable and controllable by mouse because jitsi is embedded into iframe and we can't catch any events on Stream or Meeting.

**Comment by helenvarley** [comment]
great. these things are all fine in the Meeting tool, as long as we have the individual jitsi option as a clean video image only. i'll have a look at it now on dev-app.

**Comment by helenvarley** [comment]
first thing i noticed on dev-app is that the toolbar labels have reverted to the previous "streams" (actually video) and "meeting" (actually streams) - this was corrected with issue #1308 so i will reopen that issue.

**Comment by helenvarley** [comment]
now on dev-app there is no individual jitsi option, & as far as i can see the meeting is the same as what we have on upstage.live, with all of the jitsi tools visible again to both player and audience. 

i can't find any way as a player to change the shape of the meeting, but this is not really a necessary feature for the meeting.

i think we should separate out these issues as this thread has gone into the meeting tool, when originally it was about the individual jitsi tool and specifically about needing to have it visible/audible for those entering the stage after it has been put on. 

the "meeting" and "individual" jitsi are two separate things and serve different functions for us as artist-users; i don't think it's a good idea to be trying to make them into one, it is confusing things.

**Comment by helenvarley** [comment]
ok i've created issues #1331 and #1332 to try to be clear about the requirements for jitsi Meeting, and for Individual jitsi.
@vickis please have a look and see if i've missed anything.

this issue should now continue to be only about the issue of needing Individual jitsi streams able to be seen and heard by those entering the stage after the stream has been placed on the stage.

**Comment by helenvarley** [comment]
i'll remove this from the project since it has been superceded by https://github.com/upstage-org/upstage/issues/1331, but will leave it open until the issues are all resolved.

**Comment by gloriajw** [comment]
Can you please consolidate outstanding issues to one task, in a bulleted list?

**Comment by helenvarley** [comment]
yup i will actually close this one now, because most of the time it is working as it should.
